packages/app-builder-lib/src/targets/snap.ts

```diff
@@ -71,6 +71,10 @@ export default class SnapTarget extends Target {
     }
     if (options.base != null) {
       snap.base = options.base
+      // from core22 onwards adapter is legacy
+      if (snap.base === "core22") {
+        delete appDescriptor.adapter
```

Comment:
Do we need to future-proof this for core22+? This feels too hardcoded to me. Perhaps we can extract the number from `core__` and compare `>= 22`, what are your thoughts?

Reply:
Agreed. We can change it to something like `Number(snap.base.split("core")[1]) >= 22` instead?

Reply:
That sounds perfect.